lighttpd/datadog_checks/lighttpd/lighttpd.py

```diff
@@ -89,7 +89,11 @@ def check(self, instance):
             self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.OK, tags=service_check_tags)
 
         headers_resp = r.headers
-        server_version = self._get_server_version(headers_resp)
+        full_version, server_version = self._get_server_version(headers_resp)
+
+        if full_version:
+            self.set_metadata('version', full_version)
```

Comment:
Instead of manually parsing the version, you should be able to just pass the raw value to `set_metadata` and it will figure out the rest.

Checking for the existence of the version is still a good idea though.

Reply:
See here for example: https://github.com/DataDog/integrations-core/blob/36a0d22bf7725127711d35bbd915d54522eb3154/hdfs_datanode/datadog_checks/hdfs_datanode/hdfs_datanode.py#L171

Reply:
I updated the regex to do less parsing and just pass everything after "lighttpd/" to `set_metadata`- I can't completely take it out because it is used to get the `server version` in the check. Let me know if it's better now!